### PR TITLE
feat: responsive

### PR DESCRIPTION
**Pull request by spicylemonhaha**
responsive 70%

### CODE REVIEW

File: packages/island/src/theme-default/components/BackDrop/index.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import styles from './index.module.scss';
+import type { ComponentPropsWithIsland } from 'islandjs';
+interface Props {
```
Comment (sanyuan0704) [inline packages/island/src/theme-default/components/BackDrop/index.tsx:3]: It should have a new line after the last import statement.

File: packages/island/src/theme-default/layout/Layout/useNav.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { useState } from 'react';
+export function useNav() {
```
Comment (sanyuan0704) [inline packages/island/src/theme-default/layout/Layout/useNav.ts:2]: `useNav`  and `useSidebar` should put in `logic` direcotry.

File: packages/plugin-search/src/component/index.tsx [hunk 3]
```diff
@@ -140,20 +150,28 @@ export function Search(
...
+        className={styles.searchCommand}
       >
         <span>⌘</span>
         <span>K</span>
       </div>
       {focused && query && (
         <ul
-          absolute=""
+          // absolute=""
+          // z="60"
```
Comment (sanyuan0704) [inline packages/plugin-search/src/component/index.tsx:161]: ?

File: packages/island/src/theme-default/layout/DocLayout/index.tsx [hunk 4]
```diff
@@ -12,11 +12,18 @@ export interface DocLayoutProps {
...
 export function DocLayout(props: DocLayoutProps) {
-  const { beforeDocFooter, beforeDoc, afterDoc, beforeOutline, afterOutline } =
-    props;
+  const {
+    beforeDocFooter,
+    beforeDoc,
+    afterDoc,
+    beforeOutline,
+    afterOutline,
+    isSidebarOpen
```
Comment (sanyuan0704) [inline packages/island/src/theme-default/layout/DocLayout/index.tsx:25]: I don't it's a good way to put `isSidebarOpen` as a prop.It's a little messy and not clear enough. However, `useSidebar` should can be access the state directly and make the code clean.

File: packages/island/src/theme-default/layout/Layout/useNav.ts (hunk 2 above)
Comment (spicylemonhaha) [reply packages/island/src/theme-default/layout/Layout/useNav.ts:2]: ok

File: packages/plugin-search/src/component/index.tsx (hunk 3 above)
Comment (spicylemonhaha) [reply packages/plugin-search/src/component/index.tsx:161]: delete

File: packages/island/src/theme-default/components/BackDrop/index.tsx (hunk 1 above)
Comment (spicylemonhaha) [reply packages/island/src/theme-default/components/BackDrop/index.tsx:3]: okok

**Comment by sanyuan0704** [comment]
There are still some lint problems

**Pull request merged**